core/src/main/java/org/ehcache/core/exceptions/ExceptionFactory.java

```diff
@@ -47,4 +47,30 @@ public static CacheWritingException newCacheWritingException(Exception e) {
   public static CacheLoadingException newCacheLoadingException(Exception e) {
     return new CacheLoadingException(e);
   }
+
+  /**
+   * Creates a new {@code CacheWritingException} with the provided exception as cause and a suppressed one.
+   *
+   * @param e the cause
+   * @param suppressed the suppressed exception to add to the new exception
+   * @return a cache writing exception
+   */
+  public static CacheWritingException newCacheWritingException(Exception e, Exception suppressed) {
```

Comment:
Why do we need this factory?  Wouldn't we better off just adding a fluent `supressing` method to `CacheWritingException` and having: `throw new CacheWritingException(e1).supressing(e);`

Reply:
The factory was there already. I almost deleted it and then needed the suppression. Isn't it weird to add the fluent method on these exceptions? It feels like something that should be on some `BaseEhcacheException` or something.